src/client/Microsoft.Identity.Client/Platforms/iOS/BrokerKeyHelper.cs

```diff
@@ -27,13 +26,14 @@ internal static byte[] GetRawBrokerKey(ICoreLogger logger)
             };
 
             NSData key = SecKeyChain.QueryAsData(record);
+            byte[] brokerKey;
             if (key == null)
             {
                 AesManaged algo = new AesManaged();
```

Comment:
this is disposable so please use 

```csharp
using (AesManaged algo...) 
{
// do stuff
}
```

Reply:
there are more of those around. is this important? Generally I would say the GC can handle this one ? if we are afraid of running out of resources it may be a good idea to force dispose it, otherwise I would leave it to the GC

---
In reply to: [378471346](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1627#discussion_r378471346) [](ancestors = 378471346)

Reply:
It is best practice to dispose of IDisposables, there are even static analyses rules around it (I'll look for one). There are 2 problems with not disposing of IDposable imo: 

1. GC can be slow. A second gen GC pass is invoked only about once per second. In a website running many requests per second, this can become a bottleneck.
2. Not all objects that implement IDisposable implement the entire pattern that protects against developers who forget to call `Dispose()`. If an IDisposable does not implement the full pattern, but it contains unmanaged resources, then that resource will not be cleaned up.

PS: there's a [static](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1063?view=vs-2019) analysis rule for the second point :). I'll look at enabling it.

Reply:
Yeah, my point is. how needed is it regardless? it may implement IDisposable because they have non managed resources and that's the only way to free those resources. perhaps they are able to reuse those resources if we don't dispose the object. We have had this code around for years and to my knowledge no issues so far. I would not prioritize this very high

---
In reply to: [378479075](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1627#discussion_r378479075) [](ancestors = 378479075)

Reply:
i'll make a work item for this. I agree that it's too much right now

---
In reply to: [378507050](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1627#discussion_r378507050) [](ancestors = 378507050,378479075)